src/amqp.rs

```diff
@@ -0,0 +1,78 @@
+use lapin::tcp::{OwnedIdentity, OwnedTLSConfig};
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use vector_config::configurable_component;
+
+/// Client certificate for rabbit authentication
```

Comment:
As far as I am aware, realistically RabbitMQ is the only server that we will ever talk to here - but theoretically it could apply to any AMQP 0.9.1 server.

Reply:
Then this should be AMQP, as well as properly punctuated 👍